### ISSUE TITLE
Fieldless struct constructor in patterns is treated as an identifier binding

### ISSUE THREAD

**Issue by afetisov** [intellij-rust/intellij-rust#7577]
## Environment

* **IntelliJ Rust plugin version:** 0.4.152.4004-211-nightly
* **Rust toolchain version:** 1.53.0 (53cb7b09b 2021-06-17) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.1.3 (CL-211.7628.27)
* **Operating system:** Linux 5.4.0-77-generic
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

```rust
struct Foo;

fn f(it: impl Iterator<Item=(Foo, Foo)>) {
    for (Foo, Foo) in it {}
           // ^^^ error
}
```

Error: Identifier `Foo` is bound more than once in the same pattern [E0416].

Expected: no error, because constructors of structs and enums have higher priority in pattern parsing than identifier binding.
